src/main/java/net/starlark/java/eval/StarlarkInt.java

```diff
@@ -453,10 +516,18 @@ public static StarlarkInt divide(StarlarkInt x, StarlarkInt y) throws EvalExcept
     if (y == ZERO) {
       throw Starlark.errorf("real division by zero");
     }
+
     if (x instanceof Int32 && y instanceof Int32) {
-      long xl = ((Int32) x).v;
-      long yl = ((Int32) y).v;
+      int xl = ((Int32) x).v;
+      int yl = ((Int32) y).v;
+      return StarlarkInt.of(xl / yl);
```

Comment:
This looks wrong. The result is wider than the operands because of the special case of minint / -1. Please make sure there's a test for that case.

The case below looks wrong for the same reason (minlong / -1). You need to detect this special case and fall through to the next bigger size.